perllib/Open311.pm

```diff
@@ -102,7 +102,23 @@ sub send_service_request {
             if ( my $request_id = $obj->{request}->[0]->{service_request_id} ) {
                 return $request_id unless ref $request_id;
             } elsif ( my $token = $obj->{request}->[0]->{token} ) {
-                return $self->get_service_request_id_from_token( $token );
+
+                my $service_request_id;
+                my $polling_interval = FixMyStreet->config('O311_POLLING_INTERVAL') || 2;
+                my $polling_timeout = FixMyStreet->config('O311_POLLING_TIMEOUT') || 10;
+
+                while ($polling_timeout) {
```

Comment:
Polling timeout will never be 0 if polling interval doesn't divide cleanly into polling timeout. E.g. if polling interval is 3 and polling timeout 10.

Might be better to just retry a fixed number of times, or at a fixed interval, or similar.

Reply:
How about `while ($polling_timeout > 0)`? 
I thought it might save headaches in the future if we could configure the values, but maybe that's premature optimisation!

What I've written is certainly not "the simplest thing that could possibly work", though.

Reply:
Also note it doesn't currently stop early if it does get a ID, it keeps retrying for the full number of goes.